base_gs1_barcode/models/gs1_barcode.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    This module is copyright (C) 2012-2014 Numérigraphe SARL.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+# Make it easier to divide integers and get floating point results
+from __future__ import division
+
+import re
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from openerp.tools.translate import _
+from openerp import models, fields, api, exceptions
+from openerp.tools.misc import DEFAULT_SERVER_DATE_FORMAT as DATEFMT
+
+
+class GS1Barcode(models.Model):
+    """GS1-128/GS1-Datamatrix barcode decoder API and configuration"""
+    _name = "gs1_barcode"
+    _description = __doc__
+
+    ai = fields.Char(
+        'Application Identifier',
+        size=14,
+        required=True,
+        help='The standard Application Identifier (AI)')
+    name = fields.Char('Description', required=True, translate=True)
+    length_fixed = fields.Boolean(
+        'Fixed-length Data',
+        default=True,
+        help=('Indicates whether the length of the data '
+              'for this Application Identifier is fixed or not.'))
+    length_max = fields.Integer(
+        'Maximum Data Length',
+        default=30,
+        required=True,
+        help='Maximum length of the data for this Application Identifier.')
+    length_min = fields.Integer(
+        'Minimum Data Length',
+        help='Minimum length of the data for this Application Identifier.')
+    decimal = fields.Boolean(
+        'Decimal Indicator',
+        default=False,
+        help=('Indicates whether a digit is expected before the data for this '
+              'Application Identifier to indicate the position of the decimal '
+              'point.'))
+    data_type = fields.Selection(
+        [('string', 'Any character string'), ('numeric', 'Numeric value'),
+         ('date', 'Date')],
+        'Data Type',
+        default='string',
+        required=True,
+        oldname='type')
```

Comment:
Does `oldname='type'` suffice to have the data migrated?
Otherwise can you please provide a migration script?